frontend/src/components/Cart.tsx

```diff
@@ -363,13 +391,26 @@ const Cart = ({ data, onRefetch }: Props) => {
         </Modal.Header>
         <Modal.Body>
           <Row style={{ width: '100%' }}>
-            {usableCouponData?.map((couponData, index) => (
-              <Col xs={12} md={6} key={index} style={{ padding: '3%' }}>
-                <div onClick={() => onApplyCoupon(couponData.id)} style={{ cursor: 'pointer' }}>
-                  <CouponItemTemplate data={couponData} />
-                </div>
+            {usableCouponData === undefined || usableCouponData?.length > 0 ? (
+              usableCouponData?.map((couponData, index) => (
+                <Col xs={12} md={6} key={index} style={{ padding: '3%' }}>
+                  <div onClick={() => onApplyCoupon(couponData.id)} style={{ cursor: 'pointer' }}>
+                    <CouponItemTemplate
+                      data={{
+                        name: couponData.name,
+                        discount: couponData.discount,
+                        type: couponData.type,
+                        expire_date: formatDate(couponData.expire_date),
+                      }}
+                    />
+                  </div>
+                </Col>
+              ))
+            ) : (
+              <Col xs={12} className='center'>
+                <h2>No usable coupon 😢</h2>
               </Col>
-            ))}
+            )}
```

Comment:
why isn't `usableCouponData === undefined` showing 😢

Reply:
uhh that is too make sure usableCouponData is array brefore checking sableCouponData.length

Reply:
I thought `usableCouponData?.` is enough

Reply:
> uhh that is too make sure usableCouponData is array brefore checking sableCouponData.length

wait shouldn't it be `usableCouponData === undefined && usableCouponData?.length > 0` ?

Reply:
shouldn't it be
```ts
usableCouponData && usableCouponData.length > 0
```
or
```ts
Array.isArray(usableCouponData) && usableCouponData.length > 0
```

Reply:
UG said it already fixed